packaging/src/shopify-cli.rb.liquid

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+require "formula"
+require "language/node"
+require "fileutils"
+
+class ShopifyCli{% if formulaVersion %}AT{{formulaVersion}}{% endif %} < Formula
+  desc "A CLI tool to build for the Shopify platform"
+  homepage "https://github.com/shopify/cli#readme"
+  url "{{ cliTarball }}"
+  sha256 "{{ cliSha }}"
+  license "MIT"
+  depends_on "node"
+  depends_on "ruby"
+  depends_on "git"
+
+  resource "cli-theme-commands" do
+    url "{{ themeTarball}}"
+    sha256 "{{ themeSha }}"
+  end
+
+  livecheck do
+    url :stable
+  end
+
+  def install
+    existing_cli_path = `which shopify`
+    unless existing_cli_path.empty? || existing_cli_path.include?("homebrew")
+      opoo <<~WARNING
+      We've detected an installation of the Shopify CLI at #{existing_cli_path} that's not managed by Homebrew.
+
+      Please ensure that the Homebrew line in your shell configuration is at the bottom so that Homebrew-managed
+      tools take precedence.
+      WARNING
+    end
+
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+
+    executable_path = "#{libexec}/bin/shopify{{formulaVersion}}"
+    executable_content = <<~SCRIPT
+      #!/usr/bin/env node
+
+      process.env.SHOPIFY_RUBY_BINDIR = "#{Formula["ruby"].opt_bin}"
+      process.env.SHOPIFY_HOMEBREW_FORMULA = "shopify-cli{% if formulaVersion %}@{{formulaVersion}}{% endif %}"
+
+      import("./shopify");
+    SCRIPT
+    File.write executable_path, executable_content
+    FileUtils.chmod("+x", executable_path)
+
+    bin.install_symlink executable_path
+
+    resource("cli-theme-commands").stage {
+      system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    }
+  end
+
+{%- unless formulaVersion %}
+
+  def post_install
+    message = <<~POSTINSTALL_MESSAGE
+      Congratulations, you've successfully installed Shopify CLI 3!
+
+      Global installations of Shopify CLI 3 should only be used to work on Shopify
+      themes. To learn about the changes to theme workflows in this version, refer
+      to https://shopify.dev/themes/tools/cli/migrate#workflow-changes
+
+      If you want to keep using Shopify CLI 2, then you can install it again using
+      `brew install shopify-cli@2` and run commands using the `shopify2` program
+      name (for example, `shopify2 theme push` and `shopify2 extension push`).
+      Note however that Shopify CLI 2 will be sunset in H1 2023 - specific date
+      to be announced in Q4 2022.
+    POSTINSTALL_MESSAGE
+    message.each_line { |line| ohai line.chomp }
```

Comment:
I confirmed this works by linking a local tap to Homebrew on my machine.

<img width="1512" alt="Screen Shot 2022-10-26 at 12 15 31" src="https://user-images.githubusercontent.com/6288426/197986749-8f7ef38d-4ba9-438a-be7b-8ec8b321c2ae.png">

Reply:
Note that `ohai` seems to have a limit on how many characters it prints, so I can't print a single long string.